### PR TITLE
test: added Cypress tests for Footer

### PR DESCRIPTION
**Pull request by shilpabasu22**
## Ticket Number

https://pins-ds.atlassian.net/browse/AS-

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: lpa-submissions-e2e-tests/cypress/support/common/clickFooterLink.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import {
+  accessibility,
+  cookies,
+  privacy,
+  termsAndConditions,
+} from '../PageObjects/footerPageObjects';
+
+export const clickFooterLink = (footerLink) => {
+  if (footerLink === 'Privacy') {
+    privacy().click();
```
Comment (davehlewis) [inline lpa-submissions-e2e-tests/cypress/support/common/clickFooterLink.js:10]: These can have the objects directly. Having an additional import seems to do little to add modularity here

File: lpa-submissions-e2e-tests/cypress/integration/pageFooter/pageFooter.step.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps';
+import { clickFooterLink } from '../../support/common/clickFooterLink';
+
+Given('LPA planning officer accesses the LPA Questionnaire', () => {
```
Comment (davehlewis) [inline lpa-submissions-e2e-tests/cypress/integration/pageFooter/pageFooter.step.js:4]: This might be worth being a common step - if it isn't already?

File: lpa-submissions-e2e-tests/cypress/integration/pageFooter/pageFooter.step.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps';
+import { clickFooterLink } from '../../support/common/clickFooterLink';
```
Comment (davehlewis) [inline lpa-submissions-e2e-tests/cypress/integration/pageFooter/pageFooter.step.js:2]: clickFooterLink is unlikely to be used outside of this context. Why not just have the function directly where we use it to save the cognitive load of imports?

File: lpa-submissions-e2e-tests/cypress/integration/pageFooter/pageFooter.step.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+When('they click on the {string} in the footer', (footerLink) => {
+  clickFooterLink(footerLink);
+});
+Then('the {string} correct page should be displayed', (pageUrl) => {
+  cy.verifyPage(pageUrl);
+});
+
+Then('the page title should be {string}', (pageTitle) => {
+  cy.verifyPageTitle(pageTitle);
+  cy.go('back');
```
Comment (davehlewis) [inline lpa-submissions-e2e-tests/cypress/integration/pageFooter/pageFooter.step.js:17]: Why is this back link here? additional tests don't run after this so seems superfluous?

**Review by davehlewis — CHANGES_REQUESTED** [review]

File: lpa-submissions-e2e-tests/cypress/support/PageObjects/footerPageObjects.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export const privacy = () => {
```
[the remark this answers is not in the document]
Comment (shilpabasu22) [reply lpa-submissions-e2e-tests/cypress/support/PageObjects/footerPageObjects.js:1]: What is meant by do-away? How are we interacting with the page elements?

data-cy is defined by the devs in the code and we only use them in tests. So, this has to be updated in the implementation branch.

File: lpa-submissions-e2e-tests/cypress/support/common/clickFooterLink.js (hunk 1 above)
Comment (shilpabasu22) [reply lpa-submissions-e2e-tests/cypress/support/common/clickFooterLink.js:10]: This is the industry standard for creating automation framework. If required, I can remove it but we will be deviating from the general practice and is against the SOLID principles.

File: lpa-submissions-e2e-tests/cypress/support/PageObjects/footerPageObjects.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (davehlewis) [reply lpa-submissions-e2e-tests/cypress/support/PageObjects/footerPageObjects.js:1]: As in remove them. The interactions need to be there but cypress best practice suggests to keep them in the step file. In this case the abstraction to page objects is unnecessary and just makes it harder to follow. That's why the number of page objects has been drastically reduced on the project, was so hard to follow what was happening with them and the abundance of commands

File: lpa-submissions-e2e-tests/cypress/support/common/clickFooterLink.js (hunk 1 above)
Comment (davehlewis) [reply lpa-submissions-e2e-tests/cypress/support/common/clickFooterLink.js:10]: This is against cypress best practice though. I don't profess to be an automation framework specialist, however the framework itself suggests this is not best practice.

I don't see how we are breaking SOLID principles by not doing this. Having layers of interdependency increases cognitive load when working on them, and makes changes far harder to implement and track.

File: lpa-submissions-e2e-tests/cypress/integration/pageFooter/pageFooter.step.js (hunk 4 above)
Comment (shilpabasu22) [reply lpa-submissions-e2e-tests/cypress/integration/pageFooter/pageFooter.step.js:17]: updated as per review comments

File: lpa-submissions-e2e-tests/cypress/support/common/clickFooterLink.js (hunk 1 above)
Comment (shilpabasu22) [reply lpa-submissions-e2e-tests/cypress/support/common/clickFooterLink.js]: updated as per review comments

File: lpa-submissions-e2e-tests/cypress/integration/pageFooter/pageFooter.step.js (hunk 2 above)
Comment (shilpabasu22) [reply lpa-submissions-e2e-tests/cypress/integration/pageFooter/pageFooter.step.js:15]: we can do it separately if there is a need

File: lpa-submissions-e2e-tests/cypress/integration/pageFooter/pageFooter.step.js (hunk 3 above)
Comment (shilpabasu22) [reply lpa-submissions-e2e-tests/cypress/integration/pageFooter/pageFooter.step.js]: updated as per review comments

**Review by karunak21 — APPROVED** [review]
